src/operators/courseItemOperations.ts

```diff
@@ -7,8 +7,18 @@ const checkCourseItemUser = checkUserId(CourseItem.get);
 export const create = async (courseItem: CourseItemInterface) => {
 	const hashKey = uuidv4();
 	courseItem.id = hashKey;
-	// logs uuid of new courseItem for testing
-	console.log(courseItem.name + ' id: ' + courseItem.id);
+
+	// if grade, weight, and/or date fields are empty, set to undefined
+	if (!courseItem.grade) {
+		courseItem.grade = undefined;
+	}
+	if (!courseItem.weight) {
+		courseItem.weight = undefined;
+	}
+	if (!courseItem.date) {
+		courseItem.date = undefined;
+	}
```

Comment:
Check on this and see if this is the behaviour we want, I know grade/weight being `0` would cause issues.